### PR TITLE
Update pulp client gems to allow faraday 1.x

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
I forgot to fix pulp_deb_client and I missed an epoch in the pulpcore_client package.
```
[2022-07-14T11:11:13.818Z]     package: rubygem-pulp_ansible_client-0.13.1-1.el8.noarch from el8-katello-nightly
[2022-07-14T11:11:13.818Z]       unresolved deps:
[2022-07-14T11:11:13.818Z]         rubygem(faraday) < 1
[2022-07-14T11:11:13.818Z]         (rubygem(faraday) >= 0.17 with rubygem(faraday) < 1 with rubygem(faraday) < 1.9.0)
[2022-07-14T11:11:13.818Z]     package: rubygem-pulp_container_client-2.10.3-1.el8.noarch from el8-katello-nightly
[2022-07-14T11:11:13.818Z]       unresolved deps:
[2022-07-14T11:11:13.818Z]         rubygem(faraday) < 1
[2022-07-14T11:11:13.818Z]         (rubygem(faraday) >= 0.17 with rubygem(faraday) < 1 with rubygem(faraday) < 1.9.0)
[2022-07-14T11:11:13.818Z]     package: rubygem-pulp_deb_client-2.18.0-1.el8.noarch from el8-katello-nightly
[2022-07-14T11:11:13.818Z]       unresolved deps:
[2022-07-14T11:11:13.818Z]         (rubygem(faraday) >= 0.17 with rubygem(faraday) < 1 with rubygem(faraday) < 1.9.0)
[2022-07-14T11:11:13.818Z]     package: rubygem-pulp_rpm_client-3.17.4-1.el8.noarch from el8-katello-nightly
[2022-07-14T11:11:13.818Z]       unresolved deps:
[2022-07-14T11:11:13.818Z]         rubygem(faraday) < 1
[2022-07-14T11:11:13.818Z]         (rubygem(faraday) >= 0.17 with rubygem(faraday) < 1 with rubygem(faraday) < 1.9.0)
[2022-07-14T11:11:13.819Z]     package: rubygem-pulpcore_client-1:3.18.5-1.el8.noarch from el8-katello-nightly
[2022-07-14T11:11:13.819Z]       unresolved deps:
[2022-07-14T11:11:13.819Z]         rubygem(faraday) < 1
[2022-07-14T11:11:13.819Z]         (rubygem(faraday) >= 0.17 with rubygem(faraday) < 1 with rubygem(faraday) < 1.9.0)
[2022-07-14T11:11:13.819Z]     package: rubygem-pulpcore_client-doc-1:3.18.5-2.el8.noarch from downloaded_rpms
[2022-07-14T11:11:13.819Z]       unresolved deps:
[2022-07-14T11:11:13.819Z]         rubygem-pulpcore_client = 3.18.5-2.el8
```

File: packages/katello/rubygem-pulp_certguard_client/rubygem-pulp_certguard_client.spec [hunk 1]
```diff
@@ -1,68 +1,47 @@
...
-gem unpack %{SOURCE0}
-%{?scl:EOF}
-
-%setup -q -D -T -n  %{gem_name}-%{version}
+%setup -q -n  %{gem_name}-%{version}
 
-%{?scl:scl enable %{scl} - << \EOF}
-gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
-%{?scl:EOF}
+# https://github.com/pulp/pulp-openapi-generator/pull/73
```
Comment (ehelms) [inline packages/katello/rubygem-pulp_certguard_client/rubygem-pulp_certguard_client.spec:34]: I saw this was merged. Will the next version then remove the need for these two lines below?

Comment (ekohl) [reply packages/katello/rubygem-pulp_certguard_client/rubygem-pulp_certguard_client.spec:34]: Correct, but that requires pushing new gems and that's up to the individual plugin owners. I don't want to wait for that to fix repoclosure hence this workaround.

Comment (ekohl) [reply packages/katello/rubygem-pulp_certguard_client/rubygem-pulp_certguard_client.spec:34]: In addition to that: the `gemspec_remove_dep` should fail if the exact dependency isn't found, so when it's redundant the build should fail.

**Review by ehelms — APPROVED** [review]

**Pull request merged**